pkg/kv/kvserver/batcheval/cmd_lease_transfer.go

```diff
@@ -74,7 +74,6 @@ func TransferLease(
 	prevLease, _ := cArgs.EvalCtx.GetLease()
 
 	newLease := args.Lease
-	args.Lease = roachpb.Lease{} // prevent accidental use below
```

Comment:
For reviewers: this line was intentionally removed because it was not needed and I found that it once again caused a good deal of confusion when trying to write a test. It's best not to mutate input requests, especially if they may later be exposed to testing hooks.